src/connectedk8s/azext_connectedk8s/_troubleshootutils.py

```diff
@@ -0,0 +1,567 @@
+                if(each_agent_pod.metadata.name.startswith("kube-aad-proxy")):
+                    continue
+                if each_agent_pod.status.phase != 'Running':
+                    counter = 1
+
+        # Displaying error if the arc agents are in pending state.
+        if counter:
+            print("Error: One or more Azure Arc agents are in pending state. It may be caused due to insufficient resource availability on the cluster.\n For more details on resource requirement visit 'aka.ms\\arcenabledkubernetesresourcerequirement'. \n")
+            return False, storage_space_available
+
+        return True, storage_space_available
+
+    except OSError as e:
+        storage_space_available = False
+        shutil.rmtree(filepath_with_timestamp, ignore_errors=False, onerror=None)
+
+    except Exception as e:
+        logger.warning("An expection has occured while trying to check the azure arc agents state in the cluster. Exception: {}".format(str(e)) + "\n")
+
+    return True, storage_space_available
+
+
+def check_agent_version(connected_cluster, azure_arc_agent_version):
+    try:
+
+        # If the agent version in the connected cluster resource is none skip the check
+        if(connected_cluster.agent_version is None):
+            return True
+
+        # To get user agent verison and the latest agent version
+        user_agent_version = connected_cluster.agent_version
+        current_user_version = user_agent_version.split('.')
+        latest_agent_version = azure_arc_agent_version.split('.')
+
+        # Comparing if the user version is comaptible or not
+        if((int(current_user_version[0]) < int(latest_agent_version[0])) or (int(latest_agent_version[1]) - int(current_user_version[1]) > 2)):
+            logger.warning("We found that you are on an older agent version thats not supported.\n Please visit this link to know the agent version support policy 'https://docs.microsoft.com/en-us/azure/azure-arc/kubernetes/agent-upgrade#version-support-policy'.\n")
+            return False
+
+        return True
+
+    except Exception as e:
+        logger.warning("An expection has occured while trying to check the azure arc agents version in the cluster. Exception: {}".format(str(e)) + "\n")
+
+    return True
+
+
+def executing_diagnoser_job(corev1_api_instance, batchv1_api_instance, namespace, absolute_path):
+
+    # Setting the log output as Empty
+    diagnoser_container_log = ""
+    yaml_file_path = os.path.join(absolute_path,"troubleshoot_diagnoser_job.yaml")
+    command = ["kubectl", "delete", "-f", ""]
+    command[3]=str(yaml_file_path)
+
+    # To handle the user keyboard Interrupt
+    try:
+
+        # Executing the diagnoser_job.yaml
+        config.load_kube_config()
+        k8s_client = client.ApiClient()
+
+        try:
+            utils.create_from_yaml(k8s_client, yaml_file_path)
+        except Exception as e:
+            pass
```

Comment:
We shouldn't pass here in case of exception. We shouldn't move forward and wait for job completion if the job itself is not deployed